src/store/ws/tab-report-paginated.async.store.ts

```diff
@@ -22,6 +25,50 @@ export class TabReportPaginatedAsyncStore extends BaseAsyncPaginatedDataStore<Ta
     )
   }
 
+  public getPageNo(recNo: number): number | undefined {
+    const currentPage = this.pages.find(page => {
+      if (page.data?.find(pageData => pageData._no === recNo)) {
+        return true
+      }
+    })
+
+    return currentPage ? this.pages.indexOf(currentPage) : undefined
+  }
+
+  public getRecIndex(pageNo: number, recNo: number): number | undefined {
+    let recIndex
+
+    this.pages[pageNo].data?.find((pageData, index) => {
+      if (pageData._no === recNo) {
+        recIndex = index
+
+        return true
+      }
+    })
+
+    return recIndex ?? undefined
+  }
+
+  public updateRowTags(recNo: number, tags: TTagsDescriptor) {
+    const pageNo = this.getPageNo(recNo)
+
+    if (typeof pageNo === 'number') {
```

Comment:
No need to update row if tags has difference with zones.
I think, you can check it first

Reply:
Right